.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -1,22 +1,23 @@
 
 ### Description of changes
 * Describe *what* you're changing and *why* you're doing these changes.
-* Link to impacted open issues.
 
 ### Tests
 * Describe the automated and/or manual tests executed to validate the patch.
 * Describe the added/modified tests.
 
 ### References
+* Link to impacted open issues.
 * Link to related PRs in other packages (i.e. cookbook, node).
 * Link to documentation useful to understand the changes.
 
 ### Checklist
-- [ ] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
-- [ ] Check all commits' messages are clear, describing what and why vs how.
-- [ ] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
-- [ ] Check if documentation is impacted by this change.
+- Make sure you are pointing to **the right branch**.
+- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
+- Check all commits' messages are clear, describing what and why vs how.
```

Comment:
One thing that would be really useful IMHO is to prepend a tag on the commit message related to the high level scope of it, e.g. the feature name. In this way it would be way easier to find commits by features